py-polars/polars/selectors.py

```diff
@@ -595,14 +595,16 @@ def by_dtype(
     )
 
 
-def by_name(*names: str | Collection[str]) -> SelectorType:
+def by_name(*names: str | Collection[str], match_any: bool = False) -> SelectorType:
     """
     Select all columns matching the given names.
 
     Parameters
     ----------
-    *names
+    *names : str
```

Comment:
It's actually not wrong; you type `*param` args according to the inner type, which is str. That it is iterable is already a given because of the `*`. Still, can remove👌